demo/api.py

```diff
@@ -12,6 +16,24 @@ def __init__(self, api_key: Text) -> None:
 
         self.client = MailChimp(mc_api=api_key)
 
+    @staticmethod
+    def is_valid_email(email: Text) -> bool:
+        """Use mailchimp3 helper function to validate that it will accept it as a valid
+        email"""
+        try:
+            check_email(email)
+            return True
+        except ValueError:  # purposely raised in case of invalid email
+            return False
+        except Exception as e:
+            logger.warning(
+                f"Error: exception in check_email.\n"
+                f"{type(e)} - {e}\n"
+                f"email = {email}\n"
+                f"type(email) = {type(email)}"
+            )
+            return False
```

Comment:
Yeah, good idea to str() it within the try: block before sending it to check_email.

The rest of the code already does exactly the same as what you propose, except (no pun intended! ) that I am not using a `bare except`, because that it is a bad practice with unforeseen consequences, like not being able to CTRL+C applications if you are not careful.

The message I am giving will allow us to see what was passed. Just doing logger.error(e) is not giving the actual value or type of what was passed, so it will be hard to find out where a weird value came from.

What do you think?

Reply:
Yes, definitely except Exception as e, mine was just pseudocode 😅

The reason I was thinking to keep it more general instead of printing the types, is that if we end up hitting something other than a type error, the printed information might not give us very helpful information. We could also do both.

Reply:
Ahh i see, i think i was reading it wrong. here it still gives the type of the error and the output? in that case, yes, feel free to leave as is. 👍 sorry about that